components/admin_console/team_channel_settings/channel/details/channel_details.tsx

```diff
@@ -701,15 +703,17 @@ export default class ChannelDetails extends React.PureComponent<ChannelDetailsPr
                     toPublic={isPublic}
                 />
 
-                <ChannelModeration
-                    channelPermissions={channelPermissions}
-                    onChannelPermissionsChanged={this.channelPermissionsChanged}
-                    teamSchemeID={teamScheme?.id}
-                    teamSchemeDisplayName={teamScheme?.display_name}
-                    guestAccountsEnabled={this.props.guestAccountsEnabled}
-                    isPublic={this.props.channel.type === Constants.OPEN_CHANNEL}
-                    readOnly={this.props.isDisabled}
-                />
+                {this.props.channelModerationEnabled &&
```

Comment:
Super Nit:
Can we do object destructing on these since we already have a `const {} = this.props` above? Thanks

Reply:
I know it's done a lot in this repo., but I don't see the benefit of assigning an object property to a new variable in cases like this, especially when the property is only accessed once. 

I think when [the value is accessed twice](https://perf.link/#eyJpZCI6InBkcjVvaDJuZ3lxIiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImNvbnN0IHN0YXRlID0ge2NoYW5uZWxNb2RlcmF0aW9uRW5hYmxlZDogdHJ1ZX07IiwidGVzdHMiOlt7Im5hbWUiOiJGaW5kIGl0ZW0gMTAwIiwiY29kZSI6ImNvbnN0IHtjaGFubmVsTW9kZXJhdGlvbkVuYWJsZWR9ID0gc3RhdGU7XG5pZiAoY2hhbm5lbE1vZGVyYXRpb25FbmFibGVkKSB7fVxuaWYgKGNoYW5uZWxNb2RlcmF0aW9uRW5hYmxlZCkge30iLCJydW5zIjpbNTM1MDAwLDcwODYwMDAsODAxOTAwMCw1NDA3MDAwLDU5ODIwMDAsODE2MDAwLDQ4MTMwMDAsMzUxNjAwMCw4MjczMDAwLDE5NDI2MDAwLDIzOTgwMDAsMTQ4NjAwMCw4OTAwMDAwLDExODcxMDAwLDEwNzE5MDAwLDU1OTEwMDAsNjMwNjAwMCwxMDAwNjAwMCwxMDIxNjAwMCwxMDE2NDAwMCwxNjcwNzAwMCwxODAxMTAwMCw3MzI4MDAwLDExNzQyMDAwLDE1MDM1MDAwLDE4NDcyMDAwLDE2NzQ1MDAwLDE0MTQxMDAwLDIwMzM0MDAwLDEyMzcwMDAwLDE3NzI2MDAwLDE1NDQ2MDAwLDEwMjY1MDAwLDEzNDc2MDAwLDEwNDI4MDAwLDIwMzU1MDAwLDIyMDk1MDAwLDM0NTMwMDAsNjc4NDAwMCwxNTAzNjAwMCwxODg3MjAwMCwxMjY3MzAwMCw3MDg0MDAwLDE5NDU1MDAwLDE2NzI5MDAwLDIxNTEyMDAwLDQ5NjUwMDAsMTQ3ODMwMDAsMTk2MTQwMDAsNDI1NTAwMCwyMDYxNzAwMCwxNTcwMzAwMCwxMDA3MDAwLDIyMzIxMDAwLDEyMTY4MDAwLDk3NDYwMDAsMTM3MDUwMDAsODM5MzAwMCwxNTkxNzAwMCw1NTQ4MDAwLDE5NzIwMDAwLDE5MTc3MDAwLDE1ODM3MDAwLDE4MDkxMDAwLDExNzE1MDAwLDIwMTI3MDAwLDE1MzI5MDAwLDcyODAwMCwxOTcwNzAwMCw5NjE4MDAwLDE3MDY1MDAwLDE4OTEwMDAwLDE0MzQwMDAsODUwMDAwMCwxMjIxODAwMCwxODIwMzAwMCwxMTM5OTAwMCwyNTQyNTAwMCwxMjk5NzAwMCwxMjA0MDAwLDExNzI4MDAwLDExMDQ2MDAwLDQ4MzcwMDAsMTgzNjYwMDAsMTM2NDkwMDAsMTg2NDgwMDAsOTI2MzAwMCwxNDczOTAwMCw5OTUyMDAwLDE5ODgwMDAwLDE0OTA5MDAwLDE2ODc4MDAwLDYzMDYwMDAsMTEzODUwMDAsMTI3OTkwMDAsMTMzNjUwMDAsNTI5MDAwLDk3NjEwMDAsNzU3MjAwMCwxMzY5NDAwMF0sIm9wcyI6MTIwOTI1NjB9LHsibmFtZSI6IkZpbmQgaXRlbSAyMDAiLCJjb2RlIjoiaWYgKHN0YXRlLmNoYW5uZWxNb2RlcmF0aW9uRW5hYmxlZCkge31cbmlmIChzdGF0ZS5jaGFubmVsTW9kZXJhdGlvbkVuYWJsZWQpIHt9IiwicnVucyI6WzgzMTUwMDAsNTIxNDAwMCw4NjAyMDAwLDI3MzMwMDAsODcxNzAwMCw4MzQwMDAsNDY3MTAwMCwyNDg4MDAwLDc5MjYwMDAsMTAwMCwxNTY3MDAwLDExNTIwMDAsMzQ4MjAwMCw5NTk1MDAwLDk3NDcwMDAsNDE5MjAwMCw2MDQ4MDAwLDY4NjMwMDAsOTM3MzAwMCw4MzkxMDAwLDE2MjEzMDAwLDE0NDg1MDAwLDYxNjYwMDAsMTEyNTQwMDAsMTQwMTYwMDAsMTQ4MjEwMDAsMTg2NjYwMDAsMTMxNTcwMDAsMjA0NDUwMDAsMjk1ODAwMCwxNDEwODAwMCwxMTE2MjAwMCwxMTgyMTAwMCwxMTIzNTAwMCwxNjg2NjAwMCwxODYyOTAwMCwxNzM1NjAwMCw1ODIwMDAsMTQ0MjkwMDAsMTQzNDQwMDAsMTYzMDcwMDAsMTA1MTgwMDAsNDkyOTAwMCw5NDYyMDAwLDE2MDA0MDAwLDIwNzg0MDAwLDk5MDIwMDAsMTk4NDcwMDAsOTU1NzAwMCwxMDMwNTAwMCwyMjA0OTAwMCwxMjg2NzAwMCw2MzIwMDAsMzMzOTAwMCw1ODI3MDAwLDUwODMwMDAsMTQ5NTUwMDAsODk0MzAwMCwxNjQ4NjAwMCw2MzM4MDAwLDE4MDE5MDAwLDE0MjkwMDAwLDE0NDE1MDAwLDEzODYwMDAwLDEyMTA4MDAwLDE5ODIxMDAwLDExMDgwMDAwLDEyNTUwMDAsODkyNTAwMCwyODI2MDAwLDE0MTAyMDAwLDE0ODMyMDAwLDUzNjIwMDAsMTcxMjIwMDAsMTA3OTAwMDAsMTY3OTkwMDAsOTA3NDAwMCwyMjIzNjAwMCwxNjY5NzAwMCw3MjU4MDAwLDExMDUyMDAwLDExMDcwMDAwLDkwMzcwMDAsMTgxMDkwMDAsMTEzNDEwMDAsMTI2MTkwMDAsNDMyODAwMCwxNDEyMDAwMCw5OTc5MDAwLDIwNTY1MDAwLDEyMzA3MDAwLDE5NjkzMDAwLDQxMzUwMDAsMTEzNjIwMDAsMTMzNDkwMDAsMTA2NjgwMDAsMTUyNjAwMCwxNzQ1MjAwMCw3MTgzMDAwLDEyMTU2MDAwXSwib3BzIjoxMDc5NjgwMH1dLCJ1cGRhdGVkIjoiMjAyMS0xMC0wNlQxMjo1MzozMC44NjZaIn0%3D) it's still not clearly a performance win and would use a bit more memory, but [more than twice](https://perf.link/#eyJpZCI6Im1pNmxwZWtzYWg1IiwidGl0bGUiOiJGaW5kaW5nIG51bWJlcnMgaW4gYW4gYXJyYXkgb2YgMTAwMCIsImJlZm9yZSI6ImNvbnN0IHN0YXRlID0ge2NoYW5uZWxNb2RlcmF0aW9uRW5hYmxlZDogdHJ1ZX07IiwidGVzdHMiOlt7Im5hbWUiOiJGaW5kIGl0ZW0gMTAwIiwiY29kZSI6ImNvbnN0IHtjaGFubmVsTW9kZXJhdGlvbkVuYWJsZWR9ID0gc3RhdGU7XG5pZiAoY2hhbm5lbE1vZGVyYXRpb25FbmFibGVkKSB7fVxuaWYgKGNoYW5uZWxNb2RlcmF0aW9uRW5hYmxlZCkge31cbmlmIChjaGFubmVsTW9kZXJhdGlvbkVuYWJsZWQpIHt9IiwicnVucyI6Wzc5MzUwMDAsNjEyNzAwMCw4NDU1MDAwLDU5MzgwMDAsNTQxMDAwLDYyODIwMDAsODk5MTAwMCwxMDc3ODAwMCwxNDE5MTAwMCwxNzgzODAwMCwxMDI1MDAwLDI0ODAwMDAsMTUwODYwMDAsMTA0OTEwMDAsMTEzODkwMDAsMTIwNjYwMDAsOTE1ODAwMCw4MzExMDAwLDEwNDM1MDAwLDEwOTg0MDAwLDc4MTgwMDAsMjc1MzkwMDAsMTkzODUwMDAsMTA2MDUwMDAsMTcyMzYwMDAsODYyMjAwMCw1NzY2MDAwLDcwNzQwMDAsMzEyNTAwMCwxNjkzMzAwMCwxNjAwNTAwMCwxNjY0ODAwMCwxMDAxNzAwMCw5NTQ5MDAwLDIyMDAwMDAwLDE1MTM2MDAwLDExMjcwMDAsNTg2MzAwMCw3NzkyMDAwLDg3MjYwMDAsMTk5ODQwMDAsMTA1MzAwMDAsMTA3NDQwMDAsNjUzNzAwMCwxMDEwNTAwMCwzNzczMDAwLDEwNDA3MDAwLDEwNTgwMDAwLDE4NzAyMDAwLDE0NzkyMDAwLDI0MDQ5MDAwLDExMDE4MDAwLDEwNjkwMDAwLDEyNDY1MDAwLDE1NzcwMDAwLDU3NTAwMDAsMTQ5ODYwMDAsNTc4OTAwMCwxNjkwOTAwMCw0OTYwMDAsMzg3MjAwMCwxNTcwNzAwMCw4MjExMDAwLDE3NjY0MDAwLDc3NTIwMDAsOTM1MDAwMCw3NzAzMDAwLDE1MzQxMDAwLDEyNzg0MDAwLDEyNDgzMDAwLDE3NTEyMDAwLDE5ODkwMDAsNDc2NDAwMCwxMjA2MzAwMCwxMTc0MDAwLDE3MTUzMDAwLDEwNjEyMDAwLDIwNDY1MDAwLDExNDQ2MDAwLDkxMTAwMDAsMTA2ODcwMDAsNzg4MDAwMCwyMTU5NjAwMCwxMjU0MzAwMCwxNjcyNzAwMCwxMzAyMTAwMCwxNjM5OTAwMCwxNTE0NzAwMCw5MjEyMDAwLDE3OTUyMDAwLDEwNTc4MDAwLDE5NjQ5MDAwLDEzNjMxMDAwLDc3MTAwMCwxMzc5MzAwMCwxMTI4ODAwMCwxMzkwMTAwMCwxMzE4MjAwMCw1MTQzMDAwLDEwNjgwMDAwXSwib3BzIjoxMTIwNDc4MH0seyJuYW1lIjoiRmluZCBpdGVtIDIwMCIsImNvZGUiOiJpZiAoc3RhdGUuY2hhbm5lbE1vZGVyYXRpb25FbmFibGVkKSB7fVxuaWYgKHN0YXRlLmNoYW5uZWxNb2RlcmF0aW9uRW5hYmxlZCkge31cbmlmIChzdGF0ZS5jaGFubmVsTW9kZXJhdGlvbkVuYWJsZWQpIHt9IiwicnVucyI6WzcyMDkwMDAsNDE0NzAwMCw3MjkzMDAwLDY1NjMwMDAsNDk2MDAwLDY0MTMwMDAsODkwMDAwMCw1MjY4MDAwLDEzNjQ0MDAwLDE3MTI4MDAwLDE0MjEwMDAsMjMzODAwMCwxMDA2MzAwMCwzNTI2MDAwLDEwMDM0MDAwLDEwODA5MDAwLDk2NzQwMDAsMTUyODgwMDAsODQzMjAwMCwzMDY0MDAwLDg5ODUwMDAsMjQxOTkwMDAsMTc0NTIwMDAsODk0MzAwMCwxNzI2NTAwMCwxMDc1MjAwMCw3MTAwMDAsMTU1NTIwMDAsODYyNDAwMCwxNTI1MDAwMCwxMTI0MjAwMCwxNjA4MzAwMCwxNzEwODAwMCwxMTYwMDAwMCwyMDQyNzAwMCwxNzA5MDAwMCw3OTcwMDAsMTQzOTEwMDAsODMwOTAwMCwxNDQxMDAwMCwxNTg5MTAwMCwxNTA2MjAwMCwxMjY3MDAwMCw2MDkxMDAwLDE4MjAwMDAwLDQyMjAwMDAsOTM5MjAwMCw4MzUzMDAwLDE1NjIxMDAwLDEyNzE2MDAwLDI1OTIyMDAwLDExNjk3MDAwLDkxNjUwMDAsMTI4MzIwMDAsNzg2OTAwMCw1OTg4MDAwLDE1MTE5MDAwLDE0MjkxMDAwLDE0NjQwMDAwLDQ5NjAwMCw4NzUxMDAwLDE2MzM5MDAwLDc5NDEwMDAsMTg4NDYwMDAsNzQ3MTAwMCw4NTQwMDAwLDczODYwMDAsMTMxNTQwMDAsMTM1MDkwMDAsMzcyNDAwMCwxMzk4NzAwMCw4MTg1MDAwLDExMjk5MDAwLDk0MTcwMDAsMjM2NTYwMDAsMTYzODQwMDAsMTEzMDIwMDAsMTkxODEwMDAsMTAzODgwMDAsODgzODAwMCwxODEyNTAwMCw2MTE5MDAwLDIyNzUwMDAwLDQ0NTgwMDAsMTQ2MjQwMDAsOTg2ODAwMCwxNDcxMzAwMCwxMzczODAwMCw5OTQ5MDAwLDY1NjkwMDAsNjY5NTAwMCwxODk0MzAwMCwxMDcxMjAwMCw3NTIwMDAsMTI3MDEwMDAsNTIyOTAwMCwxNDg4MTAwMCwxMDE0MDAwMCw0NTA3MDAwLDU3ODgwMDBdLCJvcHMiOjEwOTY2OTMwfV0sInVwZGF0ZWQiOiIyMDIxLTEwLTA2VDEyOjU0OjU2LjM4NloifQ%3D%3D) it seems to be a pretty consistent performance improvement.